src/components/NavBar/index.tsx

```diff
@@ -19,8 +19,13 @@ import "./NavBar.css";
 import Search from "./Search";
 import NewBookButton from "./NewBookButton";
 import { useAuthContext } from "../../store/auth/AuthContext";
+import AdminPanelButton from "./AdminPanelButton";
 
-const NavBar = () => {
+type Props = {
+  adminPage?: boolean;
+};
+
+const NavBar = ({ adminPage = false }: Props) => {
```

Comment:
On this level it's better to pass something like `showSearchBar` or something like that, and let the parent component decide whether to show it or not